Installation issue

I cannot seem to set up the project, as running gradlew build -x test returns an error with './antlrBuild'. 

running 'docker build . ' also returns the same issue.

`  > Configure project :
      Compiling Antlr Parser
      
      Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
      Use '--warning-mode all' to show the individual deprecation warnings.
      See https://docs.gradle.org/7.0/userguide/command_line_interface.html#sec:command_line_warnings
      <-------------> 0% WAITING
      > IDLE
      
      FAILURE: Build failed with an exception.
      
      * What went wrong:
      A problem occurred evaluating root project 'valgolang'.
      > A problem occurred starting process 'command './antlrBuild''
      
      * Try:
      Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
      
      * Get more help at https://help.gradle.org
      The command '/bin/sh -c gradle build -x test' returned a non-zero code: 1`

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,7 +4,7 @@ RUN apt-get update && apt-get install -y texlive texlive-latex-extra texlive-fon
                                          texlive-latex-recommended texlive-science texlive-fonts-extra tipa
 
 # Install python3 + all dependancies for manim
-RUN apt-get update && apt-get install -y python3 python3-pip sox ffmpeg libcairo2 libcairo2-dev
+RUN apt-get update && apt-get install -y python3 python3-pip sox ffmpeg libcairo2 libcairo2-dev dos2unix
 
 # Update pip for opencv dependancy
 RUN pip3 install --upgrade pip
@@ -14,6 +14,9 @@ RUN pip3 install manimlib
 
 COPY . /src/
 WORKDIR /src
+
+# Convert line endings for windows
+RUN dos2unix antlr_config/antlrBuild
 
 # Build jar file using gradle
 RUN gradle build -x test

--- a/antlr_config/antlrBuild.bat
+++ b/antlr_config/antlrBuild.bat
@@ -1,0 +1,11 @@
+set LEXER_FILE=VAlgoLangLexer.g4
+set PARSER_FILE=VAlgoLangParser.g4
+
+set TARGET_DIR=../src/main/java/antlr
+
+rmdir %TARGET_DIR% /s /q
+
+echo Compiling the lexer..
+java -jar ../lib/antlr-4.7-complete.jar %LEXER_FILE% -o %TARGET_DIR% -package antlr -no-listener -visitor -Werror 
+echo Compiling the parser..
+java -jar ../lib/antlr-4.7-complete.jar %PARSER_FILE% -o %TARGET_DIR% -package antlr -no-listener -visitor -Werror

--- a/build.gradle
+++ b/build.gradle
@@ -18,9 +18,16 @@ repositories {
 
 task buildAntlr() {
     println 'Compiling Antlr Parser'
-    exec {
-        workingDir 'antlr_config'
-        commandLine './antlrBuild'
+    if (System.getProperty('os.name').toLowerCase().contains("windows")) {
+        exec {
+            workingDir 'antlr_config'
+            commandLine 'cmd', '/c', 'antlrBuild.bat'
+        }
+    } else {
+        exec {
+            workingDir 'antlr_config'
+            commandLine './antlrBuild'
+        }
     }
 }
 
@@ -56,6 +63,7 @@ jar {
         attributes 'Main-Class': 'com.valgolang.CompilerKt'
     }
     from { configurations.compileClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
+    exclude "**/module-info.class"
 }
 
 

